[Qt6] update to 6.2.3

Neumann-A opened the pull request:
CI test run more changes incoming

> You have modified or added at least one vcpkg.json where a "license" field is missing.

Somebody wants to come up with an automatic way to add those to qt?

Neumann-A committed:
update qt to 6.2.3

Neumann-A committed:
update few details

abique commented:
Hi @Neumann-A 
Do you think you could add a feature to turn on `-optimize-size`?
It is documented here: https://github.com/qt/qtbase/blob/dev/config_help.txt#L67
I've tried adding it via the triplet: `VCPKG_CXX_FLAGS="-Os"` but Qt did not picked it up, and did build with `-O3`.
Thanks!

Neumann-A commented:
> Do you think you could add a feature to turn on -optimize-size?

no the toolchain should control that. If you find the line in qt which messes up the flags you are free to tell me and I patch it out. 
Otherwise if you correctly set `VCPKG_CXX_FLAGS` you also needed to set `VCPKG_C_FLAGS`. If you did not do that there is something wrong with your setting.

Neumann-A committed:
fix filepath.

Neumann-A committed:
use vcpkg_cmake_install

Neumann-A commented:
@autoantwort @strega-nil: There needs to be some automatic way to update the license stuff ..... i am not going to add those for qt manually considering the amount of license files and ports.

Neumann-A committed:
use the old fixup since the new one is missing TOOLS_PATH

Neumann-A committed:
more fixes

Neumann-A committed:
fix bugs

Neumann-A committed:
regex tuning

autoantwort commented:
> @autoantwort @strega-nil: There needs to be some automatic way to update the license stuff ..... i am not going to add those for qt manually considering the amount of license files and ports.

Is the license for all qt ports the same? If you know the license I can add it (Conan uses `LGPL-3.0`).

Neumann-A commented:
> Is the license for all qt ports the same? If you know the license I can add it (Conan uses LGPL-3.0).

According to the licenses in the sources it has a complicated `OR` licenscape and it is not all the same for all ports.

Neumann-A committed:
more bug fixes

Neumann-A committed:
fix qtwebengine

Neumann-A committed:
version stuff

Neumann-A committed:
ci retrigger

Neumann-A committed:
fix more unused vars

Neumann-A committed:
fix some details and silent maybe_unused warnings

Neumann-A committed:
version stuff

Neumann-A committed:
fix last warning

Neumann-A committed:
[skip actions] version stuff

ras0219-msft commented:
Sorry for the spam from actions -- the license clause is only a warning and the action will still approve without addressing it. We should definitely put the output into a details when modifying many ports at once.

Neumann-A committed:
change QT_NAMESPACE to VCPKG_QT_NAMESPACE since that is what the guidelines say for global variables.

Neumann-A committed:
version stuff

strega-nil-ms commented:
@Neumann-A I think the correct thing here is to put `"null"` as a license, since the licensing story is complex enough that we shouldn't make it simple.

I'll update the tool in the vcpkg repo with the tool that has that ability today.

strega-nil-ms commented on ports/qt/vcpkg.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "qt",
-  "version-semver": "6.2.2",
+  "version-semver": "6.2.3",
```
Can you please change all of these `version-semver`'s to `version`'s?

Neumann-A replied:
next update. I currently don't want to rerun the update script again locally which performs the change. (since this requires a removal of qtbase)

strega-nil-ms commented on ports/qtbase/cmake/qt_install_submodule.cmake:
```diff
@@ -61,40 +62,44 @@ function(qt_cmake_configure)
-            list(APPEND _qarg_OPTIONS -DQT_SYNCQT:PATH=${CURRENT_HOST_INSTALLED_DIR}/tools/Qt6/bin/syncqt.pl)
-        endif()
-        set(PERL_OPTION -DHOST_PERL:PATH=${PERL})
+    if(NOT ${PORT} MATCHES "qtbase")
```
```suggestion
    if(NOT PORT MATCHES "^qtbase")
```

strega-nil-ms commented on ports/qtbase/cmake/qt_install_submodule.cmake:
```diff
@@ -61,40 +62,44 @@ function(qt_cmake_configure)
     endif()
+    set(PERL_OPTION "-DHOST_PERL:PATH=${PERL}")
 
     if(NOT _qarg_DISABLE_NINJA)
         set(NINJA_OPTION PREFER_NINJA)
     endif()
```
This also needs to be swapped:

```suggestion
    set(ninja_option "")
    if(_qarg_DISABLE_NINJA)
        set(ninja_option WINDOWS_USE_MSBUILD)
    endif()
```

strega-nil-ms commented on ports/qtbase/portfile.cmake:
```diff
@@ -39,8 +39,7 @@ set(INPUT_OPTIONS)
         list(APPEND INPUT_OPTIONS -DINPUT_lib${_input}:STRING=)
-    elseif(_input MATCHES "(sql-sqlite)")
-        list(APPEND INPUT_OPTIONS -DINPUT_sqlite:STRING=) # Not yet used be the cmake build
+    elseif(_input MATCHES "(sql-sqlite)") # Not yet used be the cmake build
```
```suggestion
    elseif(_input MATCHES "(sql-sqlite)") # Not yet used by the cmake build
```

strega-nil-ms commented on ports/qtbase/portfile.cmake:
```diff
@@ -79,8 +78,8 @@ INVERTED_FEATURES
-if(QT_NAMESPACE)
-    list(APPEND FEATURE_OPTIONS -DQT_NAMESPACE:STRING=${QT_NAMESPACE})
+if(VCPKG_QT_NAMESPACE)
+    list(APPEND FEATURE_OPTIONS -DQT_NAMESPACE:STRING=${VCPKG_QT_NAMESPACE})
```
```suggestion
    list(APPEND FEATURE_OPTIONS "-DQT_NAMESPACE:STRING=${VCPKG_QT_NAMESPACE}")
```

strega-nil-ms reviewed (changes requested):
Minor changes requested, then I'm happy to merge :)

Neumann-A commented on ports/qtbase/cmake/qt_install_submodule.cmake:
```diff
@@ -61,40 +62,44 @@ function(qt_cmake_configure)
+
+    vcpkg_cmake_configure(
         SOURCE_PATH "${SOURCE_PATH}"
         ${NINJA_OPTION}
```
```suggestion
        ${ninja_option}
```

Neumann-A committed:
Apply suggestions from code review

Co-authored-by: nicole mazzuca <83086508+strega-nil-ms@users.noreply.github.com>

Neumann-A committed:
[skip actions] version stuff

strega-nil committed:
version-semver -> version

strega-nil-ms commented:
Merge once green!

strega-nil committed:
add license: null

Neumann-A committed:
[skip actions] version stuff

JackBoosY commented:
Can you please use `vcpkg-cmake` in qtbase?

Neumann-A commented:
> Can you please use `vcpkg-cmake` in qtbase?

It is already used? (Otherwise i would get an error) vcpkg-cmake-config isn't used yet -> next update

Neumann-A committed:
small but important prl fix

Neumann-A committed:
[skip actions] version stuff

abique commented:
> > Do you think you could add a feature to turn on -optimize-size?
> 
> no the toolchain should control that. If you find the line in qt which messes up the flags you are free to tell me and I patch it out. Otherwise if you correctly set `VCPKG_CXX_FLAGS` you also needed to set `VCPKG_C_FLAGS`. If you did not do that there is something wrong with your setting.

Hi @Neumann-A,

We can try to pass: `-DQT_USE_DEFAULT_CMAKE_OPTIMIZATION_FLAGS=ON`.

See:

https://github.com/qt/qtbase/blob/6.2.3/cmake/QtFlagHandlingHelpers.cmake#L674
https://github.com/qt/qtbase/blob/6.2.3/cmake/QtFlagHandlingHelpers.cmake#L856

Regards,
Alexandre

Neumann-A committed:
move submodule hashes into port details.

Neumann-A committed:
switch to vcpkg-cmake-config because the merge takes so long.

Neumann-A committed:
version stuff and hopefully silent bot.

Neumann-A committed:
don't let qt "cleanup" compiler flags

Neumann-A committed:
add missing dependency.

Neumann-A committed:
forgot to chain vcpkg-cmake-config

Neumann-A committed:
bit of feature setting to fix dep issues in cmake PR

Neumann-A committed:
version stuff

Neumann-A commented:
So I updated the following:
- added `-DQT_USE_DEFAULT_CMAKE_OPTIMIZATION_FLAGS=ON.`
- switched to vcpkg-cmake-config
- fixed an issues with prl files (used by vcpkg-qmake not yet tested in CI)
- added some x related FEATURES_ to fix the regression in the cmake update PR. Also fixed a missing dependency
- moved all hashes to port_details so everything is again in one location. (Needed for future LTS version feature to simply replace the file with the hashes) 
- GH actions still complains for some reason.

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into qt6_update_6.2.3

Neumann-A committed:
fix unused warnings.

Neumann-A committed:
version stuff

Neumann-A committed:
silence more warnings

Neumann-A committed:
version stuff

Neumann-A committed:
do the silencing in qttranslations correctly.

Neumann-A committed:
version stuff

Neumann-A commented:
so all MAYBE_UNUSED_WARNINGS should be silenced now.

JackBoosY commented on ports/qtbase/vcpkg.json:
```diff
@@ -1,9 +1,9 @@
+  "version": "6.2.3",
+  "description": "Qt Application Framework Base Module. Includes Core, GUI, Widgets, Networking, SQL, Concurrent and other essential qt components.",
   "homepage": "https://www.qt.io/",
+  "license": null,
```
Should be `LGPL`? See https://www.qt.io/licensing/open-source-lgpl-obligations

Neumann-A replied:
I wasnt th one adding the license here

Neumann-A committed:
silence GHA

Neumann-A committed:
version stuff

JackBoosY reviewed (approved):
LGTM.

BillyONeal commented:
@strega-nil-ms is still marked as "requested changes" but she has a more recent comment than that "merge once green" so I'm assuming she's OK with it.

BillyONeal commented:
Thanks for the update!
